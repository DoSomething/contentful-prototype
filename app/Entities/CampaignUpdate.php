<?php

namespace App\Entities;

use JsonSerializable;

class CampaignUpdate extends Entity implements JsonSerializable
{
    /**
     * Convert the object into something JSON serializable.
     *
     * @return array
     */
    public function jsonSerialize()
    {
        switch ($this->getContentType()) {
            case 'campaignUpdate':
                $type = $this->getContentType();
                $author = new Staff($this->author->entry);
                $content = $this->content;
                $socialOverride = $this->socialOverride ? new SocialOverride($this->socialOverride->entry) : null;
                break;

            case 'customBlock':
                $type = 'campaignUpdate';
                $author = [
                    'id' => null,
                    'type' => 'staff',
                    'fields' => [
                        'name' => isset($this->additionalContent['author']) ? $this->additionalContent['author'] : null,
                        'jobTitle' => isset($this->additionalContent['jobTitle']) ? $this->additionalContent['jobTitle'] : null,
                        'avatar' => null,
                    ],
                ];
                $content = "## {$this->title}\n\n {$this->content}";
                $socialOverride = null;
                break;
        }

        return [
            'id' => $this->entry->getId(),
            'type' => $type,
            'fields' => [
                'author' => $author,
                'content' => $content,
                'displayOptions' => $this->displayOptions->first(),
                'link' => $this->link,
                'socialOverride' => $socialOverride,
            ],
        ];
    }
}
