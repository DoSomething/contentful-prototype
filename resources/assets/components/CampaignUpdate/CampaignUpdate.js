import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { get } from 'lodash';

import Card from '../utilities/Card/Card';
import Embed from '../utilities/Embed/Embed';
// @see line 82 *1
// import Share from '../utilities/Share/Share';
import Byline from '../utilities/Byline/Byline';
import SponsorPromotion from '../SponsorPromotion';
import Markdown from '../utilities/Markdown/Markdown';

const CampaignUpdate = props => {
  const {
    affiliateLogo,
    author,
    id,
    closeModal,
    content,
    link,
    bordered,
    // @see line 82 *1
    // shareLink,
    // titleLink,
  } = props;

  const authorFields = get(author, 'fields', {});

  const isTweet = content && content.length < 144;

  // The affiliate logo specific text is hardcoded for OZY
  // @TODO Make this a Contentful field?
  const title = affiliateLogo ? 'See More Be More Do More' : 'Campaign Update';

  return (
    <Card
      id={id}
      className={classnames('rounded', {
        bordered,
        'affiliate-content': affiliateLogo,
      })}
      // @see line 82 *1
      // link={titleLink}
      title={title}
      onClose={closeModal}
    >
      <Markdown className={classnames('padded', { 'font-size-lg': isTweet })}>
        {content || 'Placeholder'}
      </Markdown>

      {link ? (
        <div className="padded">
          <Embed url={link} />
        </div>
      ) : null}

      <footer className="padded clearfix">
        {affiliateLogo ? (
          <SponsorPromotion className="affiliate-logo" imgUrl={affiliateLogo} />
        ) : (
          <Byline
            author={authorFields.name}
            avatar={authorFields.avatar || undefined}
            jobTitle={authorFields.jobTitle || undefined}
            className="float-left"
          />
        )}
        {/* @see line 82 *1
          <Share
            link={shareLink}
            variant="icon"
            parentSource="campaignUpdate"
            className="clear-none -right-icon"
          /> */}
      </footer>
    </Card>
  );
};

// *1:
// Temporarily sunsetting sharing and linking since
// moving campaign updates from `activity_feed` to page removes the entry as a direct descendent of
// the campaigns, thus disabling our helpers#findContentfulEntry from finding it for rendering.
// @todo implement routing for individual entries.

CampaignUpdate.propTypes = {
  id: PropTypes.string.isRequired,
  affiliateLogo: PropTypes.string,
  author: PropTypes.shape({
    id: PropTypes.string,
    type: PropTypes.string,
    fields: PropTypes.object,
  }),
  closeModal: PropTypes.func,
  content: PropTypes.string,
  link: PropTypes.string,
  // @see line 81 *1
  // shareLink: PropTypes.string.isRequired,
  // titleLink: PropTypes.string.isRequired,
  bordered: PropTypes.bool,
};

CampaignUpdate.defaultProps = {
  affiliateLogo: null,
  link: null,
  bordered: true,
  author: null,
  content: null,
  closeModal: null,
};

export default CampaignUpdate;
