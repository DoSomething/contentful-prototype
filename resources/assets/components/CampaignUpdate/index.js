export CampaignUpdate from './CampaignUpdate';

export default from './CampaignUpdate';
