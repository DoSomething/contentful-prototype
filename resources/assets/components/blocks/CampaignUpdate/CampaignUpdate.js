import React from 'react';
import gql from 'graphql-tag';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { get } from 'lodash';

import Card from '../../utilities/Card/Card';
import Embed from '../../utilities/Embed/Embed';
// @see line 81 *1
// import Share from '../../utilities/Share/Share';
import Byline from '../../utilities/Byline/Byline';
import { contentfulImageUrl } from '../../../helpers';
import TextContent from '../../utilities/TextContent/TextContent';
import AffiliatePromotion from '../../utilities/AffiliatePromotion/AffiliatePromotion';

export const CampaignUpdateBlockFragment = gql`
  fragment CampaignUpdateBlockFragment on CampaignUpdateBlock {
    id
    content
    link
    author {
      name
      jobTitle
      photo {
        url
      }
    }
    affiliateLogo {
      url
    }
  }
`;

const CampaignUpdate = props => {
  const {
    affiliateLogo,
    author,
    id,
    closeModal,
    content,
    link,
    bordered,
    // @see line 81 *1
    // shareLink,
    // titleLink,
  } = props;

  // Support both GraphQL & PHP Content API formats:
  const authorFields = author && author.fields ? author.fields : author;
  const authorPhoto = get(authorFields, 'photo.url') || undefined;

  const isTweet = content && content.length < 144;

  return (
    <Card
      id={id}
      className={classnames('rounded', { bordered })}
      // @see line 81 *1
      // link={titleLink}
      title="Campaign Update"
      onClose={closeModal}
    >
      <TextContent className={classnames('p-3', { 'text-xl': isTweet })}>
        {content || 'Placeholder'}
      </TextContent>

      {link ? (
        <div className="p-3">
          <Embed url={link} />
        </div>
      ) : null}

      <footer className="p-3 clearfix">
        {affiliateLogo ? (
          <AffiliatePromotion
            imgUrl={affiliateLogo}
            textClassName="text-gray-600"
          />
        ) : (
          <Byline
            author={authorFields.name}
            photo={
              authorPhoto
                ? contentfulImageUrl(authorPhoto, 175, 175, 'fill')
                : undefined
            }
            jobTitle={authorFields.jobTitle || undefined}
            className="float-left"
          />
        )}
        {/* @see line 81 *1
          <Share
            link={shareLink}
            variant="icon"
            parentSource="campaignUpdate"
            className="clear-none -right-icon"
          /> */}
      </footer>
    </Card>
  );
};

// *1:
// Temporarily sunsetting sharing and title links due to affects of running
// /content/management-api-scripts/2018_04_20_001_campaign_activity_feed_to_community_page
// (Moving Campaign##activity_feed blocks to a Page prevents us from finding and displaying the entries locally
// at the /block and /modal routes)
// @todo implement routing for individual entries.

CampaignUpdate.propTypes = {
  id: PropTypes.string.isRequired,
  affiliateLogo: PropTypes.string,
  author: PropTypes.shape({
    id: PropTypes.string,
    type: PropTypes.string,
    fields: PropTypes.object,
  }),
  closeModal: PropTypes.func,
  content: PropTypes.string,
  link: PropTypes.string,
  // @see line 81 *1
  // shareLink: PropTypes.string.isRequired,
  // titleLink: PropTypes.string.isRequired,
  bordered: PropTypes.bool,
};

CampaignUpdate.defaultProps = {
  affiliateLogo: null,
  link: null,
  bordered: true,
  author: null,
  content: null,
  closeModal: null,
};

export default CampaignUpdate;
