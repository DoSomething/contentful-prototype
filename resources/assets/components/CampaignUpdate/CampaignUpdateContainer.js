/* global */

import { connect } from 'react-redux';

import CampaignUpdate from './CampaignUpdate';
import { makeShareLink } from '../../helpers';
import { PHOENIX_URL } from '../../constants';

const mapStateToProps = (state, props) => {
  const linkOptions = {
    domain: PHOENIX_URL,
    slug: state.campaign.slug,
    key: props.id,
  };

  return {
    shareLink: makeShareLink('campaigns', { ...linkOptions, type: 'modal' }),
    titleLink: makeShareLink('campaigns', { ...linkOptions, type: 'blocks' }),
  };
};

export default connect(mapStateToProps)(CampaignUpdate);
