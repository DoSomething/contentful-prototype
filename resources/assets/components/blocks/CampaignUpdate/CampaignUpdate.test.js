import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import CampaignUpdate from './CampaignUpdate';

const author = {
  fields: {
    photo: 'http://example.com/avatar-aang.jpg',
    jobTitle: 'The Last Airbender',
    name: 'Aang',
  },
};

let component = shallow(
  <CampaignUpdate
    id="1234567890"
    author={author}
    content="Duis mollis, est non commodo luctus, nisi erat porttitor ligula, eget lacinia odio sem nec elit. Morbi leo risus, porta ac consectetur ac, vestibulum at eros. Sed posuere consectetur est at lobortis. Sed posuere consectetur est at lobortis. Nullam quis risus eget urna mollis ornare vel eu leo. Etiam porta sem malesuada magna mollis euismod. Cras mattis consectetur purus sit amet fermentum."
    shareLink="http://example.com/link-to-content"
    titleLink="http://example.com/link-to-content"
  />,
);

test('it generates a campaign update snapshot', () => {
  const tree = shallowToJson(component);

  expect(tree).toMatchSnapshot();
});

component = shallow(
  <CampaignUpdate
    id="1234567890"
    author={author}
    content="Duis mollis, est non commodo luctus, nisi erat porttitor ligula, eget lacinia odio sem nec elit. Morbi leo risus, porta ac consectetur ac, vestibulum at eros. Sed posuere consectetur est at lobortis. Sed posuere consectetur est at lobortis. Nullam quis risus eget urna mollis ornare vel eu leo. Etiam porta sem malesuada magna mollis euismod. Cras mattis consectetur purus sit amet fermentum."
    shareLink="http://example.com/link-to-content"
    titleLink="http://example.com/link-to-content"
    affiliateLogo="http://example.com/avatar-aang.jpg"
  />,
);

test('it generates a campaign update in affiliate mode snapshot', () => {
  const tree = shallowToJson(component);

  expect(tree).toMatchSnapshot();
});

test('it can display a campaign update as a card component', () => {
  expect(component.find('Card')).toHaveLength(1);
});
