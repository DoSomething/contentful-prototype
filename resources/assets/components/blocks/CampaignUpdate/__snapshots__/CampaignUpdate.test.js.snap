// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it generates a campaign update in affiliate mode snapshot 1`] = `
<Card
  attributes={Object {}}
  className="rounded bordered"
  id="1234567890"
  link={null}
  onClose={[Function]}
  title="Campaign Update"
>
  <TextContent
    className="p-3"
    classNameByEntry={Object {}}
    classNameByEntryDefault={null}
    styles={Object {}}
  >
    Duis mollis, est non commodo luctus, nisi erat porttitor ligula, eget lacinia odio sem nec elit. Morbi leo risus, porta ac consectetur ac, vestibulum at eros. Sed posuere consectetur est at lobortis. Sed posuere consectetur est at lobortis. Nullam quis risus eget urna mollis ornare vel eu leo. Etiam porta sem malesuada magna mollis euismod. Cras mattis consectetur purus sit amet fermentum.
  </TextContent>
  <footer
    className="p-3 clearfix"
  >
    <AffiliatePromotion
      className={null}
      imgUrl="http://example.com/avatar-aang.jpg"
      text="Powered by"
      textClassName="text-gray-600"
      title="Campaign Sponsor Logo"
    />
  </footer>
</Card>
`;

exports[`it generates a campaign update snapshot 1`] = `
<Card
  attributes={Object {}}
  className="rounded bordered"
  id="1234567890"
  link={null}
  onClose={[Function]}
  title="Campaign Update"
>
  <TextContent
    className="p-3"
    classNameByEntry={Object {}}
    classNameByEntryDefault={null}
    styles={Object {}}
  >
    Duis mollis, est non commodo luctus, nisi erat porttitor ligula, eget lacinia odio sem nec elit. Morbi leo risus, porta ac consectetur ac, vestibulum at eros. Sed posuere consectetur est at lobortis. Sed posuere consectetur est at lobortis. Nullam quis risus eget urna mollis ornare vel eu leo. Etiam porta sem malesuada magna mollis euismod. Cras mattis consectetur purus sit amet fermentum.
  </TextContent>
  <footer
    className="p-3 clearfix"
  >
    <AffiliatePromotion
      className={null}
      imgUrl="http://example.com/avatar-aang.jpg"
      text="Powered by"
      textClassName="text-gray-600"
      title="Campaign Sponsor Logo"
    />
  </footer>
</Card>
`;
